### ISSUE TITLE
Failed to find SSH identity file

### ISSUE THREAD

**Issue by IvanDelsinne** [gitahead/gitahead#140]
I have my native ssh client configured for gitlab and can perfectly use git in command line, but whenever I try to do a remote operation through gitahead (like pull/push), I get the following error: `Unable to fetch from 'origin' - failed to find SSH identity file`.

I'm on ArchLinux with latest packages.

**Comment by hackhaslam** [comment]
GitAhead is only looking for RSA and DSA keys. The version of libssh2 that we're currently using doesn't support Ed25519, but they added support for it recently. See 
https://github.com/libssh2/libssh2/issues/39. I will update to the new version of libssh2 and test it out.

**Comment by IvanDelsinne** [comment]
Wouldn't it be more favourable for gitahead to profit of the system SSH agent, like most git GUIs ?

**Comment by hackhaslam** [comment]
I don't know about *most* Git GUIs, but GitAhead doesn't shell out to git. It uses libgit2, which uses libssh2 for SSH support. There's no option to use the system OpenSSH.

**Comment by IvanDelsinne** [comment]
Makes sense. Should I also recompile gitahead after recompiling a version of libssh2 that supports ed?

**Comment by hackhaslam** [comment]
GitAhead also needs to be taught to look for the correct identity file. You may be able to work around it by explicitly specifying an identity file in ~/.ssh/config like this:

    Host *
      IdentityFile ~/.ssh/id_ed25519

If you're building GitAhead from source, then you should be able to update the libssh2 submodule to the tip of master and then just rebuild GitAhead. I'm going to start looking at this now, so I may have something better for you to test soon.

**Comment by IvanDelsinne** [comment]
I'm using the AUR package for now, thanks I'll try this ssh/config solution.

**Comment by hackhaslam** [comment]
The proposed workaround of setting IdentityFile in the config file isn't going to work. GitAhead also needs to be taught to prompt for the passphrase to decrypt the Ed25519 key. The existing path for detecting encrypted RSA keys isn't sufficient.

**Comment by hackhaslam** [comment]
Of course, I didn't look closely enough at the libssh2 PR to notice that it depends on OpenSSL 1.1.1. We would need to upgrade OpenSSL too. It's going to take a bit longer to look into changing that.

**Comment by pssoft7** [comment]
This doesn't seem to include support for Mac OS.

**Comment by hackhaslam** [comment]
It works for me on macOS. I've been using it continuously ever since I made the change. What doesn't work for you?

**Comment by pssoft7** [comment]
Does this work for you when both rsa/dsa and ed25519 exist?

**Comment by hackhaslam** [comment]
Yes, I have both rsa and ed25519 keys and config file entries to choose the correct key for each host.

**Comment by pssoft7** [comment]
I see. Will the ed25519 key still work if the rsa key is no longer valid?

**Comment by hackhaslam** [comment]
It should. I don't know why it wouldn't.
